### PR TITLE
WIP: Refactor to make linting more strict

### PR DESCRIPTION
**Pull request by Sqrcz**
Hey 👋 

Based on the discussion here: https://github.com/themesberg/flowbite-vue/discussions/200

This is my proposal for updating linting rules... (details in comments below).

This is Draft PR as if changes are accepted it'll be required to actually implement new rules... this can be done as part of this commit (but it would make it very big) or as separated piece of work after this one is accepted.

When updating linting dependencies I've also updated existing ones as well.

Pleas let me know if this set of rules is acceptable... If yes I'll prepare next one to implement necessary changes to align with those rules.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
   env: {
     browser: true,
     es2021: true,
     node: true,
   },
-  parser: 'vue-eslint-parser',
-  extends: ['eslint:recommended', 'plugin:vue/essential', 'plugin:@typescript-eslint/recommended'],
+  extends: [
+    'eslint:recommended',
+    'standard',
```
Comment (Sqrcz) [inline .eslintrc.js:11]: this is a set of rules aligned with what previously was part or `rules` option

File: .eslintrc.js [hunk 2]
```diff
@@ -1,25 +1,29 @@
...
+    'plugin:vue/vue3-recommended',
```
Comment (Sqrcz) [inline .eslintrc.js:12]: main change based on the discussion https://github.com/themesberg/flowbite-vue/discussions/200
it introduces stricter rules based on official `eslint-plugin-vue`

File: .eslintrc.js [hunk 3]
```diff
@@ -1,25 +1,29 @@
...
+    'plugin:vue-scoped-css/vue3-recommended',
```
Comment (Sqrcz) [inline .eslintrc.js:13]: this allows for linting `<style>` parts of SFC's

File: .eslintrc.js [hunk 4]
```diff
@@ -1,25 +1,29 @@
...
+    '@vue/eslint-config-typescript/recommended'
```
Comment (Sqrcz) [inline .eslintrc.js:14]: this is Vue dedicated version of `plugin:@typescript-eslint/recommended`

File: .eslintrc.js [hunk 5]
```diff
@@ -1,25 +1,29 @@
...
-    semi: ['error', 'never'],
-    'vue/multi-word-component-names': 'off',
     'comma-dangle': ['error', 'always-multiline'],
+    'linebreak-style': ['error', 'unix'],
     'no-multiple-empty-lines': 'error',
     'object-curly-spacing': ['error', 'always'],
-    'vue/no-v-model-argument': 'off',
+    'sort-imports': ['error', { ignoreDeclarationSort: true }],
+    'vue/block-order': ['error', { order: ['template', 'script', 'style'] }],
+    'vue/component-name-in-template-casing': ['error', 'kebab-case'],
```
Comment (Sqrcz) [inline .eslintrc.js:27]: `parser` and `plugins` were not needed for everything to work correctly

rules `quotes` and `semi` are now part of the `standard` ruleset

`'vue/multi-word-component-names': 'off',` - we want it to be `ON` and it is as part of `'plugin:vue/vue3-recommended',`
`vue/no-v-model-argument` this was Vue 2 rule.

File: package.json [hunk 6]
```diff
@@ -42,30 +42,28 @@
...
   },
   "devDependencies": {
-    "@types/lodash-es": "4.17.9",
+    "@rushstack/eslint-patch": "^1.4.0",
+    "@types/lodash-es": "^4.17.9",
     "@types/node": "20.6.0",
-    "@typescript-eslint/eslint-plugin": "6.7.0",
-    "@typescript-eslint/parser": "6.7.0",
     "@vitejs/plugin-vue": "4.3.4",
+    "@vitest/coverage-v8": "^0.34.4",
```
Comment (Sqrcz) [inline package.json:49]: code coverage wasn't working with `c8` but now it does with `"@vitest/coverage-v8"`

**Comment by Sqrcz** [comment]
@cogor 

I've updated this PR with changes to **Paragraph** and **Heading** components... it covers components as well as docs for them. Let me know if that is ok... and I'll work on the rest of the components.

Also... this PR will affect most of the files... so if you have some work in progress let's figure out how to best avoid conflicts ;)

**Comment by cogor** [comment]
That is ok, but need to fix an error in docs  :)

**Comment by Sqrcz** [comment]
good catch, thanks

cool... I'll continue with other components

**Comment by Sqrcz** [comment]
Hey... this is still WIP... I have few more components to go through... ;)

**Comment by cogor** [comment]
I know, I merged it into `fwb-rename` branch, I'll help you out and we'll make faster

**Comment by Sqrcz** [comment]
Hey @cogor ... I've finished all and it's on the same [branch](https://github.com/Sqrcz/flowbite-vue/tree/refactor-make-linting-more-strict)

I've also created PR [here](https://github.com/themesberg/flowbite-vue/pull/231)

**Pull request merged**